File: composeApp/src/toDestroyJava/src/model/AlimIndic.java
package model;

import java.util.ArrayList;
import javax.swing.DefaultListModel;


public enum AlimIndic{
	ALL("All", 999),
	PED("Pédiatrique",0),
	NEUT("Stérilisé",1),
	PHYS("Physiologique",2),
	SEN("Sénior",3),
	CALM("Stress félin",4),
	OBES("Obésité",5),
	GESTATION("Gestation",6),
	SONDE("Sonde",7),
	LACT("Lactation",8),
	CROISSANCE("Croissance",9),
	
	DENT("Hygiéne Buccodentaire",11),
	DIAB("Diabète",12),
	
	INSHEP("Insuffisance Hépatique",25),
	HYPO("Hypoallergénique",26),
	ART("Soutien Articulaire",27),
	MRC("Soutien de la fonction rénale",28),

	CONV("Convalescence",30),

	MBAUF("MBAUF",32),
	URO("Urolithiase",33),
	DERM("Affections cutanées",34),
	GI("Affections gastro-intestinales",35),
	CAR("Affections cardiaques",36),
	END("Affections endocriniennes",37),
	IPE("Insufisance pancréatique",38),
	DISTRU("Dissolution struvites",39),
	REDSTRU("Réduction struvites",40),
	REDURA("Réduction urates",41),
	REDOXA("Réduction oxalates",42),
	REDCYST("Réduction cystines",43),
	ACT("Sport",45),
	AUTRE("", 44)
	//ANOR("Anorexie",10),
	//CONS("Constipation",13),
		//MEG("Mégacolon",14),
		//COL("Colite",15),
		//DIAR("Diarrhée",16),
		//MICI("MICI",17),
		//MALD("Malabsorption, Maldigestion",18),
		
		//GAST("Gastrite",19),
		//PANC("Pancreatite",20),
		//ENTEX("Enteropathie exudative",21),
		//CHOLANG("Cholangite, Cholestase",22),
		//SHUNT("Shunt Porto-Systémique",23),
		//ENCEPHEP("Encéphalose Hépatique",24),
	//LIP("Lipidose hépatique",31),
	//CANC("Cancer",29),
		;
		


	private String name = "";
	private int coef = 1;

	//Constructeur
	AlimIndic(String name, int coef){
	this.name = name;
	this.coef=coef;
	}
	public static AlimIndic StringToGroup(String sz){
		AlimIndic r=AlimIndic.AUTRE;
		for (AlimIndic ga: AlimIndic.values()){
			if (ga.nameToString().equals(sz)){
				r=ga;
			}
			
		}
		return r;
	}
	public static AlimIndic IntToGroup(int sz){
		AlimIndic r=AlimIndic.PHYS;
		for (AlimIndic ga: AlimIndic.values()){
			if (ga.getCoef()==sz){
				r=ga;
			}
			
		}
		return r;
	}
	
	public static DefaultListModel<String>  getAllIndic(){
		  DefaultListModel<String> list = new DefaultListModel<>();
		for (AlimIndic elem:AlimIndic.values()){
			list.addElement(elem.nameToString());
		}
		return list;
	}
	public static ArrayList<AlimIndic> valuesExcept() {
		
		ArrayList<AlimIndic>es=new ArrayList<AlimIndic>();
		for (AlimIndic e:AlimIndic.values()) {
			if (e!=AlimIndic.ALL) {
				es.add(e);
			}
				
		}
		return es;
	}
	
	
	public String nameToString(){
	return name;
	}

	public int getCoef(){
		  return coef;
		}
	public String getCoefStr(){
		if (coef<10){
			  return "0"+coef;
		}else{
		  return ""+coef;}
		}
	}

File: composeApp/src/commonMain/kotlin/fr/vetbrain/vetnutri_mp/Enumer/AlimIndic.kt
package fr.vetbrain.vetnutri_mp.Enumer

import fr.vetbrain.vetnutri_mp.Data.Labelable

enum class AlimIndic(val coef: Int, override val label: String) : Labelable {
    ALL(999, "All"),
    PED(0, "Pédiatrique"),
    NEUT(1, "Stérilisé"),
    PHYS(2, "Physiologique"),
    SEN(3, "Sénior"),
    CALM(4, "Stress félin"),
    OBES(5, "Obésité"),
    GESTATION(6, "Gestation"),
    SONDE(7, "Sonde"),
    LACT(8, "Lactation"),
    CROISSANCE(9, "Croissance"),
    DENT(11, "Hygiéne Buccodentaire"),
    DIAB(12, "Diabète"),
    INSHEP(25, "Insuffisance Hépatique"),
    HYPO(26, "Hypoallergénique"),
    ART(27, "Soutien Articulaire"),
    MRC(28, "Soutien de la fonction rénale"),
    CONV(30, "Convalescence"),
    MBAUF(32, "MBAUF"),
    URO(33, "Urolithiase"),
    DERM(34, "Affections cutanées"),
    GI(35, "Affections gastro-intestinales"),
    CAR(36, "Affections cardiaques"),
    END(37, "Affections endocriniennes"),
    IPE(38, "Insufisance pancréatique"),
    DISTRU(39, "Dissolution struvites"),
    REDSTRU(40, "Réduction struvites"),
    REDURA(41, "Réduction urates"),
    REDOXA(42, "Réduction oxalates"),
    REDCYST(43, "Réduction cystines"),
    ACT(45, "Sport"),
    AUTRE(44, "");

    companion object {
        fun byCoef(coef: Int): AlimIndic = entries.find { it.coef == coef } ?: PHYS

        fun isPresent(indic: AlimIndic): Boolean = entries.contains(indic)

        fun byName(name: String): AlimIndic =
                entries.find { it.label.equals(name, ignoreCase = true) } ?: AUTRE

        fun valuesExcept(): List<AlimIndic> = entries.filter { it != ALL }

        /**
         * Récupère une indication à partir d'une chaîne, en essayant plusieurs formats possibles.
         * Cette méthode nettoie automatiquement la chaîne d'entrée (crochets, guillemets, espaces)
         * et gère l'insensibilité à la casse.
         *
         * @param value La chaîne à convertir en indication
         * @return L'indication correspondante (AUTRE si aucune correspondance n'est trouvée)
         */
        fun getFromString(value: String): AlimIndic {
            // Nettoyer la chaîne d'entrée
            val cleanedValue = value.replace("[", "").replace("]", "").replace("\"", "").trim()

            if (cleanedValue.isEmpty()) {
                return AUTRE
            }

            // Vérifier si c'est un nom d'énumération (OBES, URO, etc.) - insensible à la casse
            try {
                val upperCaseValue = cleanedValue.uppercase()
                return valueOf(upperCaseValue)
            } catch (e: Exception) {
                // Pas un nom d'énumération valide
            }

            // Vérifier si c'est un label par nom - déjà insensible à la casse via byName
            val byLabel = byName(cleanedValue)
            if (byLabel != AUTRE) {
                return byLabel
            }

            // Vérifier si c'est un coefficient numérique
            val coef = cleanedValue.toIntOrNull()
            if (coef != null) {
                return byCoef(coef)
            }

            // Aucune correspondance trouvée
            return AUTRE
        }
    }

    override fun toString() = label
}
